src/shared/treeview/nodes/awsResourceNode.ts

```diff
@@ -0,0 +1,8 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export interface AWSResourceNode {
+    getArn(): string
```

Comment:
can this just be added to `AWSTreeNodeBase` instead?

Reply:
I thought about that, but `AWSTreeNodeBase` is used for all explorer nodes. All AWS resources have an ARN, but not every explorer node represents AWS resource.

Reply:
I agree we wouldn't want this in AWSTreeNodeBase. We may want to extend certain nodes even further, or in different ways than a AwsResourceNode approach. Longer term I wonder if we'll want to do something along the lines of having nodes indicate which "capabilities" or "features" they have. At this time, I think AwsResourceNode is a reasonable approach.

Reply:
The other option is to this a child of `AWSTreeNodeBase` which we do in JetBrains and would prefer

Reply:
Extending/deriving classes/implementations might be better left for functionality instead of representing classifications of data.

An example would be if we introduced a "self-refreshing" node. This would likely derive from `AWSTreeNodeBase`. Some resource types might be self refreshing, while others wouldn't. At that point, you wouldn't extend `AWSResourceNode` from both, without having to introduce duplicate types.

The way `AWSResourceNode` is set up in this PR seems reasonable, and low cost to adjust if we realize a need.

Reply:
Keeping `AWSResourceNode` as additional interface based on the discussions in this PR.

> Would prefer resource node is a child of AWSTreeNodeBase then we can get rid of the filter in package.json and adding new nodes only requires changing one place

I don't think that would help. As @awschristou mentioned, the `where` filtering is pretty limited. The current implementation identifies node types using `TreeNode.contextValue` which is just a string we set to indicate the type.